source/kernel/src/drivers/emb_display.rs

```diff
@@ -0,0 +1,212 @@
+//! Simulated display driver
+//!
+//! This is an early attempt at a "frame buffer" style display driver.
+//!
+//! This implementation is sort of a work in progress, it isn't really a *great*
+//! long-term solution, but rather "okay for now".
+//!
+//! A framebuffer of pixels is allocated for the entire display on registration.
+//! This could be, for example, 400x240 pixels.
+//!
+//! The driver will then allow for a certain number of "sub frames" to be requested.
+//!
+//! These sub frames could be for the entire display (400x240), or a portion of it,
+//! for example 200x120 pixels.
+//!
+//! Clients of the driver can draw into the sub-frames that they receive, then send
+//! them back to be rendered into the total frame. Any data in the client's sub-frame
+//! will replace the current contents of the whole frame buffer.
+//!
+//! The current server assumes 8 bits per pixel, which is implementation defined for
+//! color/greyscale format (sorry).
+
+use crate::{
+    comms::oneshot::Reusable,
+    registry::{Envelope, KernelHandle, RegisteredDriver, ReplyTo},
+    Kernel,
+};
+use embedded_graphics::{
+    pixelcolor::{Gray8, GrayColor},
+    prelude::*,
+};
+use mnemos_alloc::containers::HeapArray;
+use uuid::Uuid;
+
+////////////////////////////////////////////////////////////////////////////////
+// Service Definition
+////////////////////////////////////////////////////////////////////////////////
+
+// Registered driver
```

Comment:
nit: maybe worth including a doc comment for the services which expect to be implemented by platform impls explaining that there's no actual server task implemented in the kernel lib? like
```suggestion
/// Registered driver type for the `EmbDisplay` service.
///
/// This module provides an implementation of the client for this service, but
/// not the server. A server implementing this service must be provided by the
/// hardware platform implementation.
```